plugins/fundchannel.c

```diff
@@ -326,14 +328,25 @@ static struct command_result *fundchannel_start(struct command *cmd,
 	json_out_end(ret, '}');
 	json_out_finished(ret);
 
-	/* FIXME: as a nice feature, we should check that the peer
-	 * you want to connect to is connected first. if not, we should
-	 * connect and then call fundchannel start!  */
 	return send_outreq(cmd, "fundchannel_start",
 			   fundchannel_start_done, tx_abort,
 			   fr, take(ret));
 }
 
+static struct command_result *connect_to_peer(struct command *cmd,
+                                              struct funding_req *fr)
+{
+	struct json_out *ret = json_out_new(NULL);
+
+	json_out_start(ret, NULL, '{');
+	json_out_addstr(ret, "id", node_id_to_hexstr(tmpctx, fr->id));
+	json_out_end(ret, '}');
+	json_out_finished(ret);
+
+	return send_outreq(cmd, "connect", fundchannel_start, tx_abort,
```

Comment:
there's no reason to call `tx_abort` here in the failure case, we can just pipe the failure from connect back to the client.

Reply:
That's what `tx_abort` will do, it's useless but does not seem to hurt ? This avoids another failure callback which does essentially the same

Reply:
that's not true, `tx_abort` makes an extra rpc call to lightningd; `forward_error` (plugins/libplugin.c#L283) is what you want

Reply:
Thanks